File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.go
/*
Time - O(n)
Space - O(1)
*/
func twoSum(nums []int, target int) []int {
    lo:=0
    hi:=len(nums)-1
    for lo<hi {
        if nums[lo]+nums[hi]==target {
            return []int{lo+1, hi+1}
        }
        if nums[lo]+nums[hi]<target {
            lo++
        }else {
            hi--
        }
    }
    return []int{-1,-1}
}

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        lo = 0
        hi = len(numbers)-1
        while lo<hi:
            if numbers[lo]+numbers[hi]==target:
                return [lo+1, hi+1]
            if numbers[lo]+numbers[hi]<target:
                lo+=1
            else:
                hi-=1
        return [-1,-1]
        